### 다중회귀분석

In [8]:
data <- lm(rating~., data=attitude)

In [9]:
data


Call:
lm(formula = rating ~ ., data = attitude)

Coefficients:
(Intercept)   complaints   privileges     learning       raises     critical  
   10.78708      0.61319     -0.07305      0.32033      0.08173      0.03838  
    advance  
   -0.21706  


In [11]:
summary(data)


Call:
lm(formula = rating ~ ., data = attitude)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.9418  -4.3555   0.3158   5.5425  11.5990 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.78708   11.58926   0.931 0.361634    
complaints   0.61319    0.16098   3.809 0.000903 ***
privileges  -0.07305    0.13572  -0.538 0.595594    
learning     0.32033    0.16852   1.901 0.069925 .  
raises       0.08173    0.22148   0.369 0.715480    
critical     0.03838    0.14700   0.261 0.796334    
advance     -0.21706    0.17821  -1.218 0.235577    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 7.068 on 23 degrees of freedom
Multiple R-squared:  0.7326,	Adjusted R-squared:  0.6628 
F-statistic:  10.5 on 6 and 23 DF,  p-value: 1.24e-05


1. Residuals : Data의 잔차
2. Adjusted R-squared : 결정계수 -> 설명력
3. p-value : 유의확률 -> 타당성
4. Pr(>|t|) : 독립변수가 종속변수에게 주는 영향 -> 영향력
5. Std. Error : 표준오차
6. t value : 비표준화 계수

#### 분석 해석
각 개별 변수들의 유의확률을 확인한 결과 'complaints' 변수만 유의미한 영향을 주는 것을 확인할 수 있다.

유의수준을 0.05라고 가정했을 때, p-value는 1.24e-05이므로 0.05보다 훨씬 작은 값을 가진다.
그러므로 해당 회귀모형은 통계적으로 타당하다고 할 수 있다.

통계적으로 타당하지만 결정계수는 0.6628로 약 66%의 설명력밖에 가지지 못하는 것을 알 수 있다.
실제 유의미성을 지니기엔 다소 부족한 결정계수이다.